chunk/chunk_store_test.go

```diff
@@ -146,3 +146,44 @@ func diff(want, have interface{}) string {
 	})
 	return "\n" + text
 }
+
+func TestBigBuckets(t *testing.T) {
+	scenarios := []struct {
+		from, through, dailyBucketsFrom model.Time
+		buckets                         []string
+	}{
+		{
+			from:             model.TimeFromUnix(0),
+			through:          model.TimeFromUnix(0).Add(3*24*time.Hour) - 1,
+			dailyBucketsFrom: model.TimeFromUnix(0).Add(1 * 24 * time.Hour),
+			buckets:          []string{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "d1", "d2"},
+		},
+		{
+			from:             model.TimeFromUnix(0),
+			through:          model.TimeFromUnix(0).Add(3*24*time.Hour) - 1,
+			dailyBucketsFrom: model.TimeFromUnix(0).Add(2*24*time.Hour) - 1, // Only the day matters for the epoch start time, not the time.
+			buckets:          []string{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "d1", "d2"},
+		},
+		{
```

Comment:
I don't understand this example well enough to suggest documentation. It looks like there's an open/closed interval thing going on that I'm not getting.